book/src/pong_tutorial/pong_tutorial_02.md

```diff
@@ -347,35 +347,43 @@ fn on_start(&mut self, world: &mut World) {
 }
 ```
 
-Okay! We've defined our `Paddle` component, and created two entities which have 
-`Paddle`, `Transform`, `MeshHandle`, and `Material` components. When our game 
-starts, we'll register the `Paddle` component and then add the left and right 
+Okay! We've defined our `Paddle` component, and created two entities which have
+`Paddle`, `Transform`, `MeshHandle`, and `Material` components. When our game
+starts, we'll register the `Paddle` component and then add the left and right
 paddles to the world, along with a camera.
 
 Let's run this and see what happens. On my machine I get a panic that reads:
 
 ```
-No component with the given id. Did you forget to register the component with 
+No component with the given id. Did you forget to register the component with
 `World::register::<ComponentName>()`?
 ```
 
-It looks like we're missing at least one component registration. In addition to 
-components we define ourselves, Amethyst has a lot of internal systems and 
-components it uses to keep things running. For simplicity, these have been 
-wrapped up into "Bundles" which include related components, systems, and 
-resources. We can add these to our Application using the `with_bundle` method, 
+It looks like we're missing at least one component registration. In addition to
+components we define ourselves, Amethyst has a lot of internal systems and
+components it uses to keep things running. For simplicity, these have been
+wrapped up into "Bundles" which include related components, systems, and
+resources. We can add these to our Application using the `with_bundle` method,
 and in fact we already have one of these in `main.rs`: the `RenderBundle`.
 
-As it turns out, the components we're missing are `Transform` and 
-`LocalTransform`, and we can add those with the `TransformBundle`, which will 
+As it turns out, the components we're missing are `Transform` and
+`LocalTransform`, and we can add those with the `TransformBundle`, which will
 also add the `TransformSystem` for working with those components:
 
 ```rust,ignore
-let mut game = Application::build("./", Pong)?
-    .with_bundle(TransformBundle::new())? //Add this bundle
+let mut renderer = None;
+let game = Application::build("./", Pong)?
+    .with_bundle(TransformBundle::new())?
     .with_bundle(RenderBundle::new())?
-    .with_local(RenderSystem::build(pipe, Some(config))?)
-    .build()?;
+    .world(|world| {
```

Comment:
You can add `RenderSystem` directly to the world inside the closure.

Reply:
Except you don't add systems to the world.  You add it to the `DispatcherBuilder`.

Reply:
Add a system to the world?

Reply:
Ah. Yeah. Forget about it 😜

Reply:
Can you make it so the closure would return `RenderSystem` and the it would be added to the `Application`?

Reply:
My suggestion, see below :)